src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4902,7 +4902,9 @@ bool StorageReplicatedMergeTree::getFakePartCoveringAllPartsInPartition(const St
         auto zookeeper = getZooKeeper();
         delimiting_block_lock = allocateBlockNumber(partition_id, zookeeper);
         right = delimiting_block_lock->getNumber();
-        mutation_version = queue.getCurrentMutationVersion(partition_id, right);
+        // Make sure we cover all parts in partition. In rare cases there might be parts with
+        // mutation version greater than current block number.
```

Comment:
AFAIK, it should never happen due to https://github.com/ClickHouse/ClickHouse/blob/20b9af7b2653a7e58bb00e3fa1b9058f4ae78cb5/src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp#L2016-L2017

Reply:
Yeah, I fail to construct a minimal reproduceable case for this, but it somehow happens.

```
Part 20210419_0_5263_999999999 intersects previous part 20210419_0_3820_6_5263. It is a bug.
```

and 

```
Part 20210218_0_12_999999999_10 intersects previous part 20210218_0_0_0_727
```

Mutation gets out of sync with the max block number.

However, it seems reasonable to have drop/detach partition to cover all parts, so that this ill-behaved case can be fixed by detach partition and attach partition without data loss.